projects/core/src/product/services/product-page-meta.resolver.ts

```diff
@@ -45,10 +46,25 @@ export class ProductPageMetaResolver
     filter(Boolean)
   );
 
+  /**
+   * @deprecated since 3.1 we'll use the contentPageResolver going forward
```

Comment:
Correct. We start deprecating the constructor, but cannot yet replace the hard-coded robots as it would introduce a breaking change. I also don't like to bring in feature flags because of the boilerplace and additional migration steps.